packages/datepicker/src/react/use-key-events.ts

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { format, endOfWeek, startOfWeek, add, sub, getMonth } from 'date-fns'
+import { useDebounceCallback } from '@pluralsight/ps-design-system-util'
+
+interface UseKeyEvents {
+  selected: Date
+  months: number[]
+  handleForwardClick: React.MouseEventHandler<HTMLButtonElement>
+  handleBackClick: React.MouseEventHandler<HTMLButtonElement>
+}
+export const useKeyEvents = ({
+  selected,
+  months,
+  handleForwardClick,
+  handleBackClick
+}: UseKeyEvents) => {
+  const [focusedDate, setFocusedDate] = React.useState<Date | undefined>(
+    selected
+  )
+  const updateFocused = () => {
+    focusedDate &&
+      document
+        .querySelector<HTMLButtonElement>(
+          `[aria-label="${format(focusedDate, 'EEE LLL dd yyyy')}"]`
+        )
+        ?.focus()
+  }
+
+  const focusNext = useDebounceCallback(updateFocused, 100)
```

Comment:
Please help me understand why this needs debounced?

Reply:
Switching between months causes a rerender such that we can't focus a button immediately in the new month view. Debouncing for 100ms worked better than waiting on RAF.